### ISSUE TITLE
False positive with unpacking-non-sequence

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#161]
Originally reported by: **Vincent Duvert (BitBucket: [VinDuv](http://bitbucket.org/VinDuv), GitHub: @VinDuv?)**

---

With the following code:

```
#!python
import os

def f():
    if os.getpid() % 2:
        return None

    return (1, 2, 3)

def test():
    result = f()

    if result is None:
        return

    a, b, c = result

    print "a = %d, b = %d, c = %d" % (a, b, c)
```

pylint complains that "a, b, c = result" may try to unpack a None result even though this is not possible with this code.

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/161
